Assets/Project/Modules/Enemies/Slime/Scripts/SlimeRefactor/SlimeMindEnemy.cs

```diff
@@ -56,11 +58,10 @@ public Transform GetPlayerTransform()
         {
             return _attackTarget;
         }
-        public void InitAfterSpawn(SlimeMediator slimeMediator)
+        public void InitAfterSpawn(SlimeMediator slimeMediator,EnemyPatrolling.PatrolType patrolType)
         {
-            
-            
-            if(_patrolType == EnemyPatrolling.PatrolType.FixedWaypoints){slimeMediator.SetWayPoints(_wayPoints);}
+            _patrolType = patrolType;
+            _slimeMediator = slimeMediator;
```

Comment:
I think that InitAfterSpawn() should only receive the mediator as a parameter. 
And then default the _patrolType to variable to the "None" value.

Reply:
And the consumers who call SetPatrollingWaypoints() should also then call EnterPatrollingMode()